### PR TITLE
Feature/healthcheck behavior changes

### PR DESCRIPTION
**Pull request by frank0631**
General improvement of healthchecks. 
Healthcheck emails now sent even when Fatal or Error level healthchecks fail.
More robust healthcheck test

gocd http client has longer timeout and skips verify of credentials

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -15,7 +15,7 @@ type Config struct {
 	logOutput    io.Writer
 	logLevel     log.Level
 
-	pgHost     string
+	PgHost     string
```
Comment (tanderegg) [inline config/config.go:18]: Can we make sure these are all consistent?  Either all "Pg" or all "pg".

File: gocd/gocd.go [hunk 2]
```diff
@@ -86,6 +88,18 @@ type Server struct {
 	Password string
 }
 
+// client returns a http client with longer timeout and skip verify
+func client() *http.Client {
+	timeout := time.Duration(120 * time.Second)
+	transCfg := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
```
Comment (tanderegg) [inline gocd/gocd.go:95]: Why is it necessary to skip verification?

File: healthcheck/healthcheck_test.go [hunk 3]
```diff
@@ -62,7 +62,35 @@ func TestEvaluatingErrorsChecks(t *testing.T) {
...
+		t.Fail()
+	}
+}
+
+func TestPreformingingErrorsChecks(t *testing.T) {
+	cxn := database.GetPGConnection(conf.DBURI())
+	healthChecks, _ := ReadHealthCheckYAMLFromFile("healthchecksErrors.yml")
+	results, HCerrs := healthChecks.PreformHealthChecks(cxn)
+
+	// log.Info("Spew healthcheck results")
```
Comment (tanderegg) [inline healthcheck/healthcheck_test.go]: Can we either remove these commented out lines, or change them to log.Debug if they might be needed?

File: healthcheck/templateHealthcheck.go [hunk 4]
```diff
@@ -35,8 +49,38 @@ const TemplateHealthcheck = `
...
+
+// SubjectHealthcheck creates a subject for healthcheck email
+func SubjectHealthcheck(name string, dbName string, hostname string, level string, errors int, fatal bool) string {
+
+	hcName := name
+	if name == "" {
+		hcName = "healthchecks"
+	}
+
+	//subjectStr := fmt.Sprintf("%s for %s on %s at %s level",
```
Comment (tanderegg) [inline healthcheck/templateHealthcheck.go]: Can we remove this commented out line?

File: gocd/gocd.go (hunk 2 above)
Comment (frank0631) [inline gocd/gocd.go:95]: the ssl cert for the production gocd is invalid. See 'curl -k'

File: rhobot.go [hunk 5]
```diff
@@ -263,6 +267,11 @@ func healthcheckRunner(config *config.Config, healthcheckPath string, reportPath
 				}
 			}
 		}
+	}
 
+	// Bad Exit
+	if HCerrs != nil {
+		//spew.Dump(HCerrs)
```
Comment (tanderegg) [inline rhobot.go]: Here too, can we remove the commented line or switch it to Debug logging?

**Comment by tanderegg** [comment]
@frank0631 Made a few minor comments, otherwise looks good, thanks Frank!

File: config/config.go (hunk 1 above)
Comment (frank0631) [inline config/config.go:18]: I've been making them public as needed. I'll make them all public now.

**Comment by tanderegg** [comment]
@frank0631 Thanks much, looks good!

**Pull request merged**